File: react-native/library/src/starPlanet/components/TspRedeemCardGrid.js
import React from 'react';
import { Pressable, StyleSheet, Text, View } from 'react-native';
import { withTheme } from '../utils/shared';

/**
 * TspRedeemCardGrid – Redeem item cards.
 */
export function TspRedeemCardGrid({
  items = [],
  availablePoints = 0,
  frozen = false,
  disabled = false,
  theme,
  onRedeem,
}) {
  const t = withTheme(theme);

  return (
    <View style={styles.grid}>
      {frozen ? (
        <View
          style={[
            styles.banner,
            {
              backgroundColor: t.activeFill,
              borderColor: t.warning,
            },
          ]}
        >
          <Text style={[styles.bannerText, { color: t.textPrimary }]}>今日已冻结，暂不可兑换</Text>
        </View>
      ) : null}
      {items.map((item) => {
        const cost = Number(item.cost) || 0;
        const insufficient = availablePoints < cost;
        const blocked = frozen || disabled || insufficient;
        return (
          <Pressable
            key={String(item.id ?? item.name)}
            disabled={blocked}
            onPress={blocked ? undefined : () => onRedeem?.(item)}
            style={[
              styles.card,
              {
                backgroundColor: t.surfaceRaised,
                borderColor: t.borderDefault,
              },
              blocked && styles.blocked,
            ]}
          >
            <Text style={styles.icon}>{item.icon || '🎁'}</Text>
            <Text style={[styles.name, { color: t.textPrimary }]} numberOfLines={2}>
              {item.name || ''}
            </Text>
            <Text style={[styles.cost, { color: t.brandPrimary }]}>{`${cost} 分`}</Text>
            {insufficient && !frozen ? (
              <Text style={[styles.hint, { color: t.danger }]}>积分不足</Text>
            ) : null}
          </Pressable>
        );
      })}
    </View>
  );
}

const styles = StyleSheet.create({
  grid: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    gap: 12,
    justifyContent: 'space-between',
  },
  banner: {
    width: '100%',
    borderWidth: 1,
    borderRadius: 14,
    paddingHorizontal: 12,
    paddingVertical: 10,
  },
  bannerText: { fontWeight: '800', fontSize: 13 },
  card: {
    width: '48%',
    borderWidth: 1,
    borderRadius: 18,
    padding: 14,
    alignItems: 'flex-start',
  },
  blocked: { opacity: 0.5 },
  icon: { fontSize: 28 },
  name: { fontWeight: '800', fontSize: 14, marginTop: 6 },
  cost: { fontWeight: '900', marginTop: 4 },
  hint: { fontSize: 12, marginTop: 4 },
});

File: flutter/library/lib/src/tsp_redeem_card_grid.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

class TspRedeemItem {
  const TspRedeemItem({
    this.id,
    this.name = '',
    this.icon,
    this.cost = 0,
  });

  final Object? id;
  final String name;
  final String? icon;
  final num cost;
}

/// Redeem item cards.
class TspRedeemCardGrid extends StatelessWidget {
  const TspRedeemCardGrid({
    super.key,
    this.items = const [],
    this.availablePoints = 0,
    this.frozen = false,
    this.disabled = false,
    this.theme = StarPlanetTheme.sky,
    this.onRedeem,
  });

  final List<TspRedeemItem> items;
  final num availablePoints;
  final bool frozen;
  final bool disabled;
  final StarPlanetTheme theme;
  final ValueChanged<TspRedeemItem>? onRedeem;

  @override
  Widget build(BuildContext context) {
    return LayoutBuilder(
      builder: (context, constraints) {
        final gap = 12.0;
        final singleCol = constraints.maxWidth <= 480;
        final cardWidth = singleCol ? constraints.maxWidth : (constraints.maxWidth - gap) / 2;

        return Wrap(
          spacing: gap,
          runSpacing: gap,
          children: [
            if (frozen)
              SizedBox(
                width: constraints.maxWidth,
                child: Container(
                  padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 8),
                  decoration: BoxDecoration(
                    color: theme.activeFill,
                    borderRadius: BorderRadius.circular(12),
                  ),
                  child: Text(
                    '今日已冻结，暂不可兑换',
                    style: TextStyle(color: theme.textPrimary, fontWeight: FontWeight.w800, fontSize: 13),
                  ),
                ),
              ),
            ...items.map((item) {
              final cost = item.cost;
              final insufficient = availablePoints < cost;
              final blocked = frozen || disabled || insufficient;
              return SizedBox(
                width: cardWidth,
                child: Opacity(
                  opacity: blocked ? .5 : 1,
                  child: GestureDetector(
                    onTap: blocked ? null : () => onRedeem?.call(item),
                    child: Container(
                      constraints: const BoxConstraints(minHeight: 120),
                      padding: const EdgeInsets.all(14),
                      decoration: BoxDecoration(
                        color: theme.surfaceRaised,
                        borderRadius: BorderRadius.circular(18),
                        border: Border.all(color: theme.borderDefault, width: 2),
                      ),
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          Text(item.icon ?? '🎁', style: const TextStyle(fontSize: 28)),
                          const SizedBox(height: 8),
                          Text(
                            item.name,
                            style: TextStyle(color: theme.textPrimary, fontWeight: FontWeight.w800),
                          ),
                          const SizedBox(height: 4),
                          Text(
                            '$cost 分',
                            style: TextStyle(color: theme.brandPrimary, fontWeight: FontWeight.w900),
                          ),
                          if (insufficient && !frozen) ...[
                            const SizedBox(height: 4),
                            Text('积分不足', style: TextStyle(color: theme.danger, fontSize: 12)),
                          ],
                        ],
                      ),
                    ),
                  ),
                ),
              );
            }),
          ],
        );
      },
    );
  }
}
